File: csharp/MudbaseShowcase.Events/Services/QrTokenGenerator.cs
namespace MudbaseShowcase.Events.Services;

/// <summary>
/// A random, unguessable single-use check-in code. Not a security credential in the
/// cryptographic sense (this is a demo ticketing app, not a payments system) - direct port of
/// ../../web/src/lib/utils.ts's `generateQrToken` (`crypto.randomUUID().replace(/-/g, "")`):
/// `Guid.NewGuid().ToString("N")` produces the identical shape - 32 lowercase hex characters, no
/// dashes - from the same 122 bits of random UUID entropy.
/// </summary>
public static class QrTokenGenerator
{
    public static string Generate() => Guid.NewGuid().ToString("N");
}

File: java/src/main/java/dev/mudbase/showcase/events/support/QrTokenGenerator.java
package dev.mudbase.showcase.events.support;

import java.util.UUID;

/**
 * A random, unguessable single-use check-in code for a new booking. Not a security credential in
 * the cryptographic sense (this is a demo ticketing app, not a payments system) - collision
 * resistance against a random UUID's 122 bits of entropy is more than sufficient for a QR
 * check-in token. Mirrors the reference web app's `generateQrToken()` (../web/src/lib/utils.ts):
 * a random UUID with its hyphens stripped.
 */
public final class QrTokenGenerator {

  private QrTokenGenerator() {}

  public static String generate() {
    return UUID.randomUUID().toString().replace("-", "");
  }
}
